### PR TITLE
[#1866143, #1866141] define and register springcloud's actions and explorer node.

### PR DESCRIPTION
**Pull request by wangmingliang-ms**

### CODE REVIEW

File: azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/IActionsContributor.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.ide.common;
+
+import com.microsoft.azure.toolkit.ide.common.action.AzureActionManager;
+
+public interface IActionsContributor {
```
Comment (shenqianjin) [inline azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/IActionsContributor.java:10]: Contributor ? what's that?

File: azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/IActionsContributor.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    default void registerActions(AzureActionManager am) {
+    }
+
+    default void registerGroups(AzureActionManager am) {
+    }
+
+    default void registerHandlers(AzureActionManager am) {
```
Comment (shenqianjin) [inline azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/IActionsContributor.java:17]: It looks these three methods should belong to its parameter `AzureActionManager`

File: azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/IActionsContributor.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    default int getZOrder() {
```
Comment (shenqianjin) [inline azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/IActionsContributor.java:20]: is there any special purpose to add `Z` character before `Order` ?

File: azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/action/ResourceCommonActions.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    public static final String START = "action.resource.start";
+    public static final String STOP = "action.resource.stop";
+    public static final String RESTART = "action.resource.restart";
+    public static final String REFRESH = "action.resource.refresh";
+    public static final String DELETE = "action.resource.delete";
+    public static final String OPEN_PORTAL_URL = "action.resource.open_portal_url";
+    public static final String SHOW_PROPERTIES = "action.resource.show_properties";
+    public static final String DEPLOY = "action.resource.deploy";
+    public static final String CREATE = "action.resource.create";
+    public static final String OPEN_URL = "action.open_url";
```
Comment (shenqianjin) [inline azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/action/ResourceCommonActions.java:30]: missing "resource". not `action.resource.url` ?

File: azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/action/ResourceCommonActions.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    @Override
+    public void registerActions(AzureActionManager am) {
+        this.registerCommand(am);
```
Comment (shenqianjin) [inline azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/action/ResourceCommonActions.java:34]: I think `command` is not easy to understand here

File: azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/action/ResourceCommonActions.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+        final Consumer<IAzureResource<?>> openPortalUrl = s -> {
+            final Runnable runnable = () -> am.getAction(OPEN_URL).handle(s.portalUrl());
+            tm.runInBackground(AzureOperationBundle.title("common|resource.open_portal_url", s.name()), runnable);
+        };
+        final Function<IAzureResource<?>, IView.Label> openPortalUrlView = s -> {
+            final String description = AzureOperationBundle.title("common|resource.open_portal_url", s.name()).toString();
+            return new IView.Label.Static("Open In Portal", "/icons/action/portal.svg", description);
+        };
+        am.registerAction(OPEN_PORTAL_URL, new Action<>(openPortalUrl, openPortalUrlView));
```
Comment (shenqianjin) [inline azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/action/ResourceCommonActions.java:83]: It looks there are a few different among these actions.
I suggest you a mind to extract these codes as a `yaml` or `xml` configuration:

common.actions:
--start:
----name: "Start"
----label: "Start"
----progressMessage: "Start %s (%s)"
----icon: "/icon/action/start.png"
--stop:
----name: "Stop"
----label: "Stop"
----progressMessage: "Stop %s (%s)"
----icon: "/icon/action/stop.png"
springcloud.actions:
--ref: ${common.actions}
--deploy:
----name: "Deploy"
----label: "Deploy"
----progressMessage: "Deploy spring clound app (%s)"
----icon: "/icon/action/deploy.png"
----actionPerformed: com.xxx.xxx.Deployable.deploy()

File: azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/IExplorerContributor.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.ide.common;
+
+import com.microsoft.azure.toolkit.ide.common.component.Node;
+
+public interface IExplorerContributor {
+    Node<?> getNode();
```
Comment (Flanker32) [inline azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/IExplorerContributor.java:11]: -> getModule()

**Pull request merged**